sdk/storage/AzureStorageBlob/Source/DataStructures/StorageStreamUploader.swift

```diff
@@ -0,0 +1,540 @@
+            } else {
+                completion()
+            }
+            return
+        }
+        let defaultGroup = DispatchGroup()
+        let dispatchGroup = group ?? defaultGroup
+        for _ in blockList {
+            dispatchGroup.enter()
+            next(inGroup: dispatchGroup) { _, _ in
+                if let delegate = self.delegate {
+                    let progress = BlobUploadProgress(bytes: self.progress, totalBytes: self.fileSize)
+                    delegate.uploader(self, didUpdateWithProgress: progress)
+                }
+            }
+        }
+        dispatchGroup.notify(queue: DispatchQueue.main) {
+            // Once all blocks are done, commit block list
+            self.commit { result, _ in
+                switch result {
+                case let .success(data):
+                    let progress = BlobUploadProgress(bytes: self.progress, totalBytes: self.fileSize)
+                    if let delegate = self.delegate {
+                        delegate.uploader(self, didFinishWithProgress: progress)
+                    } else {
+                        completion()
+                    }
+                case let .failure(error):
+                    if let delegate = self.delegate {
+                        delegate.uploader(self, didFailWithError: error)
+                    }
+                }
+            }
+        }
+    }
+
+    public func commit(
+        requestId: String? = nil,
+        transactionalContentMd5: Data? = nil,
+        transactionalContentCrc64: Data? = nil,
+        inGroup _: DispatchGroup? = nil,
+        then completion: @escaping HTTPResultHandler<BlobProperties>
+    ) {
+        // Construct URL
+        let urlTemplate = "/{container}/{blob}"
+        let pathParams = [
+            "container": containerName,
+            "blob": blobName
+        ]
+        let url = client.url(forTemplate: urlTemplate, withKwargs: pathParams)
+
+        // Construct parameters
+        var queryParams = [
+            ("comp", "blocklist")
+        ]
+        if let timeout = options.timeout { queryParams.append(("timeout", String(timeout))) }
+
+        // Construct headers
+        var headers = HTTPHeaders([
+            .contentType: "application/xml; charset=utf-8",
+            .apiVersion: client.options.apiVersion
+        ])
+        let leaseAccessConditions = options.leaseAccessConditions
+        let modifiedAccessConditions = options.modifiedAccessConditions
+        let cpk = options.cpk
+
+        if let transactionalContentMd5 = transactionalContentMd5 {
+            headers[.contentMD5] = String(data: transactionalContentMd5, encoding: .utf8)
+        }
+
+        if let transactionalContentCrc64 = transactionalContentCrc64 {
+            headers[.contentCRC64] = String(data: transactionalContentCrc64, encoding: .utf8)
+        }
+
+        if let requestId = requestId { headers["x-ms-client-request-id"] = requestId }
```

Comment:
Same comment here - we should probably use the HTTPHeader and StorageHTTPHeader values here (and below) instead of raw strings.

Reply:
Yep, done.